app/Http/Transformers/CustomFieldsetsTransformer.php

```diff
@@ -0,0 +1,46 @@
+<?php
+namespace App\Http\Transformers;
+
+use App\Models\CustomFieldset;
+use App\Models\CustomField;
+use App\Models\AssetModel;
+use Illuminate\Database\Eloquent\Collection;
+use App\Helpers\Helper;
+use Gate;
+
+class CustomFieldsetsTransformer
+{
+
+    public function transformCustomFieldsets (Collection $fieldsets, $total)
+    {
+        $array = array();
+        foreach ($fieldsets as $fieldset) {
+            $array[] = self::transformCustomFieldset($fieldset);
+        }
+        return (new DatatablesTransformer)->transformDatatables($array, $total);
+    }
+
+    public function transformCustomFieldset (CustomFieldset $fieldset)
+    {
+        $fields = $fieldset->fields;
+        $models = $fieldset->models;
+        $modelsArray = array();
+        $totalModels = 0;
+        foreach ($models as $model)
+        {
+            $modelArray = array();
+            $modelArray["id"] = $model->id;
+            $modelArray["name"] = $model->name;
```

Comment:
Style-only, but wouldn't it be cleaner to do
```
$modelsArray[] = [
'id' => $model->id,
'name' => $model->name
];
```
here?  alternatively $modelArray = $model->only(['id','name']) may work.  Would avoid having to count and simplify this block.

Alternatively, you could take advantage of the fact that laravel returns a collection from the hasMany, and do something like
```
$models->each(function($item) use ($modelsArray) {
    // Same code here
}
```